Foundation/NSDictionary.swift

```diff
@@ -156,13 +156,44 @@ public class NSDictionary : NSObject, NSCopying, NSMutableCopying, NSSecureCodin
     }
     
     public func copyWithZone(zone: NSZone) -> AnyObject {
-        NSUnimplemented()
+        if self.dynamicType === NSDictionary.self {
+            // NSDictionary is immutable; just return ourself
+            return self
+        } else if self.dynamicType === NSMutableDictionary.self {
+            // Otherwise, create a new NSDictionary object
+            
+            // TODO: speed up?
+            var keys = [AnyObject]()
+            var values = [AnyObject]()
+            for (key, value) in self {
+                keys.append(key)
+                values.append(value)
+            }
+            
+            return NSDictionary(objects: values, forKeys: keys as! [NSObject])
+        } else {
+            NSRequiresConcreteImplementation()
```

Comment:
the intent of the NSMutableDictionary case you have written is probably correct for here.